### DISCUSSION TITLE
Are backups encrypted?

### DISCUSSION THREAD

**Discussion by TBloop** [Authenticator-Extension/Authenticator#602] [Q&A] [answered]
Hi there,

Sorry if this isnt the right place to ask this - I couldn't find anywhere else. 

I have been using the automatic backup feature and I didn't know if it was safe to continue doing so if someone gained access to my dropbox, for instance.

I wanted to know if the "password protected backup" stores the authenticator secret in an encrypted manner? When I open the .json file it creates, I can see various things in plaintext, and I wasn't sure if any of this could be used to re-create the one time passwords. I see there are two fields - 'hash' and 'secret'. Are these useless without the password?

Thank you!

**Comment by mymindstorm** [comment, marked as answer]
Backups are encrypted by default, and you will see some red warning text if you disable it. To check this setting go to `Storage & Backup` -> `Dropbox` -> `Encrypted` dropdown should show `Yes`.

In terms of the backup format: the `hash` field on individual accounts is actually a random UUID, and it is just used as an internal ID. The `secret` field is encrypted with the value in `key` which is encrypted by your actual password and hashed with argon2 (all encryption is done with AES-256). All other account metadata is unencrypted.

The data that is not encrypted does not allow attackers to reconstruct OTPs, but it may give them your email or the website that the account is used on.

**Comment by TBloop** [comment]
Thanks. I appreciate the explanation.

I dont know if this would be too complicated, but wouldn't it provide a
little more security to just encrypt the entire thing? That way you dont
give up usernames or even what websites you might have passwords to.

**Reply by mymindstorm** [reply]
It would be more secure, but this is made somewhat complicated because our code is built around this format.

**Comment by bittner** [comment]
I'm a bit confused. :confused:  The project website states: [Encrypted backups not supported](https://authenticator.cc/docs/en/otp-backup)

![image](https://user-images.githubusercontent.com/665072/150687688-41f35c4e-aa82-4678-b384-87c8b1c59a7f.png)

What does this refer to, exactly? The restore process for backups from other apps? :thinking: 

Or does the website need to be updated?

**Reply by mymindstorm** [reply]
When using the interoperable backup format, encrypted backups are not supported. In the extension there is a "download password-protected backup" button for an encrypted backup.

**Reply by Sneezry** [reply]
**Encrypted backups not supported** to save as Standard Backup Format.

**Reply by ashutoshsaboo** [reply]
Are there plans to support optional encryption (maybe add a toggle in options if someone wants it to be encrypted/not) with standard backup format (the interoperable one) too? This would be really great to have! @Sneezry @mymindstorm

**Reply by Sneezry** [reply]
There isn't a "standard" way to encrypt backups and different clients use their own way to encrypt data. So, I don't think it is possible to support encryption for interoperable backups, currently. A workaround is to use a password-protected ZIP file that contains the standard backup file to transfer data among 2FA clients.
